libraries/AP_HAL_ChibiOS/hwdef/CIPHERWINGF4/hwdef.dat

```diff
@@ -0,0 +1,181 @@
+# hw definition file for processing by chibios_pins.py
+# Cipherwing F4 with STM32F405
+# mpu6000 / bmi270 / icm20689 IMU options, bmp280 barometer,
+# 7456 series OSD and NO sdcard
+
+BOARD_NAME CIPHERWINGF4
+
+MCU STM32F4xx STM32F405xx
+
+# board ID. See Tools/AP_Bootloader/board_types.txt
+APJ_BOARD_ID AP_HW_CIPHERWINGF4
+
+
+# crystal frequency
+OSCILLATOR_HZ 8000000
+
+STM32_ST_USE_TIMER 5
+
+# flash size
+FLASH_SIZE_KB 1024
+FLASH_RESERVE_START_KB 64
+
+# order of I2C buses (moved to I2C1)
+I2C_ORDER I2C1
+
+# order of UARTs (include USART2 and UART5)
+SERIAL_ORDER OTG1 USART1 USART2 USART3 USART6 UART4 UART5
+
+# ---------- ADC ----------
+# VBAT stays on PC2
+PC2 BAT_VOLT_SENS ADC1 SCALE(1)
+
+# RSSI moved to PC1 (per request)
+PC1 RSSI_IN ADC1
+define BOARD_RSSI_ANA_PIN 1
+
+# (REMOVED to avoid conflict) Current sensor on PC1:
+# PC1 BAT_CURR_SENS ADC1 SCALE(1)
+
+# ---------- PWM outputs remapped to match requested INAV-style layout ----------
+# M1 PB0  TIM3_CH3
+# M2 PB1  TIM3_CH4
+# M3 PC8  TIM8_CH3
+# M4 PC9  TIM8_CH4
+# M5 PB14 TIM1_CH2N   (F405-valid replacement for TIM12_CH1)
+# M6 PA8  TIM1_CH1
+# M7 PB15 TIM1_CH3N   (F405-valid replacement for TIM12_CH2)
+# M8 PB7  TIM4_CH2
+PB0  TIM3_CH3   TIM3  PWM(1) GPIO(50)
+PB1  TIM3_CH4   TIM3  PWM(2) GPIO(51)
+PC8  TIM8_CH3   TIM8  PWM(3) GPIO(52)
+PC9  TIM8_CH4   TIM8  PWM(4) GPIO(53)
+PB14 TIM1_CH2N  TIM1  PWM(5) GPIO(54)
+PA8  TIM1_CH1   TIM1  PWM(6) GPIO(55)
+PB15 TIM1_CH3N  TIM1  PWM(7) GPIO(56)
+PB7  TIM4_CH2   TIM4  PWM(8) GPIO(57)
+
+# ---------- IMU on SPI1 with CS=PA4 ----------
+PA4 MPU6000_CS CS
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+
+# ---------- I2C1 on PB8/PB9 (pullups). (Freed PB10/PB11 for USART3 ALTs) ----------
+PB8  I2C1_SCL I2C1 PULLUP
+PB9  I2C1_SDA I2C1 PULLUP
+
+# Keep USART3 available on PB10/PB11 via ALTs if needed
+PB10 USART3_TX USART3 ALT(1)
+PB11 USART3_RX USART3 ALT(1)
+PB10 USART3_TX USART3 ALT(4)
+PB11 USART3_RX USART3 ALT(4)
+
+# ---------- SPI2 / SD card REMOVED (per request) ----------
+# (PB15 SPI2_MOSI SPI2)
+# (PB14 SPI2_MISO SPI2)
+# (PB13 SPI2_SCK  SPI2)
+# (PB12 SDCARD_CS CS)
+
+# ---------- UARTs ----------
+PA10 USART1_RX USART1
+PA9  USART1_TX USART1
```

Comment:
Put the standard protocol functions expected for these

Reply:
okay still waiting on good image to be included in readdme  before I detail review...I took it as meaning ALL the uarts need protocol defines...thanks for clarifying